numpy/_core/src/multiarray/item_selection.c

```diff
@@ -384,6 +384,12 @@ PyArray_PutTo(PyArrayObject *self, PyObject* values0, PyObject *indices0,
         return NULL;
     }
 
+    if (PyArray_Size((PyObject *)self) == 0) {
```

Comment:
With the check at this position the following code will fail even though it is valid in current numpy:
```
np.put(a, [], 2, mode='wrap')
```
Maybe move the check down a bit and replace it with 
```
if ( (ni > 0) && (PyArray_Size((PyObject *)self) == 0) ) {
```
(`ni` is the number of indices specified)

Reply:
Workinng on it. Thanks for the suggestion.